ch_pipeline/hfb/analysis.py

```diff
@@ -936,3 +937,59 @@ def _ensure_list(x):
         y = [x]
 
     return y
+
+
+class HFBMedianSubtraction(task.SingleTask):
+    """Subtracting weighted median along beam axis to remove fluctuations in the
+    data induced by temperature fluctuations in East and West receiver huts."""
+
+    def process(self, stream):
+        """Subtract weighted median of the data along beam axis from the data.
+        A binary mask (0 when the weight is zero, and 1 for non-zero weights)
+        is used in weighted median
+
+        Parameters
+        ----------
+        stream : containers.HFBData
+            Container with HFB data and weights.
+
+        Returns
+        -------
+        out : containers.HFBData
+            Container with HFB data and weights.
+        """
+
+        # Extract data from container
+        data = stream.hfb[:]
+        weight = stream.weight[:]
+
+        # Change data to numpy array, so that it can be reshaped
+        if isinstance(data, mpiarray.MPIArray):
+            data = data.local_array
+            weight = weight.local_array
+
+        # make a mask of non-zero weights
+        mask = weight != 0
+
+        # Generate binary weight
+        binary_weight = mask.astype(float)
+
+        # Change the order of axes in data and mask arrays, as weighted median
+        # is calculated along the last axis
+        data_s = np.swapaxes(data, 2, 3)
+        binary_weight = np.swapaxes(binary_weight, 2, 3)
+
+        # Calculate weighted median (to exclude flagged data) along beam axis:
+        median = weighted_median.weighted_median(data_s, binary_weight)
+
+        # Subtract weighted median along all beams from the data
+        diff = data - median[:, :, None, :]
+
+        # Create container to hold output
+        out = containers.HFBData(stream)
+
+        # Place diff in output container
+        out.hfb[:] = diff[:]
+        out.weight[:] = weight[:]
```

Comment:
```suggestion
        out.hfb[:] = diff
        out.weight[:] = weight
```
Not 100% sure, but I think the `[:]` is not necessary